pkg/api/v1/types.go

```diff
@@ -494,6 +494,13 @@ func ConvertInternalUserToExternal(internalUser *kubermaticv1.User, includeSetti
 			ID:                internalUser.Name,
 			Name:              internalUser.Spec.Name,
 			CreationTimestamp: NewTime(internalUser.CreationTimestamp.Time),
+			DeletionTimestamp: func() *Time {
+				if internalUser.DeletionTimestamp != nil {
+					deletionTimestamp := NewTime(internalUser.DeletionTimestamp.Time)
+					return &deletionTimestamp
+				}
+				return nil
+			}(),
```

Comment:
You could expose this as a global function and then use it in all the handlers to convert timestamps. :)

Reply:
I am sure someone will eventually refactor this :)